tests/flows/test_server.py

```diff
@@ -55,8 +56,264 @@ def mocked_runtask(*args, **kwargs):
     # Verify that GDS and SVG files were generated and returned.
     assert os.path.isfile('build/gcd/job0/export/0/outputs/gcd.gds')
 
+###########################
+@pytest.mark.eda
+@pytest.mark.quick
+def test_gcd_server_partial(gcd_chip):
+    '''Basic sc-server test: Run a local instance of a server, and build the GCD
+       example using loopback network calls to that server.
+
+       This test runs a partial flowgraph on the remote server.
+    '''
+
+    # Start running an sc-server instance.
+    os.mkdir('local_server_work')
+    srv_proc = subprocess.Popen(['sc-server',
+                                 '-port', '8081',
+                                 '-nfs_mount', './local_server_work',
+                                 '-cluster', 'local'])
+    time.sleep(3)
+
+    # Mock the _runstep method.
+    old__runtask = gcd_chip._runtask
+    def mocked_runtask(*args, **kwargs):
+        if args[0] == 'import':
+            old__runtask(*args)
+        else:
+            gcd_chip.logger.error('Non-import step run locally in remote job!')
+    gcd_chip._runtask = Mock()
+    gcd_chip._runtask.side_effect = mocked_runtask
+
+    # Ensure that klayout doesn't open its GUI after results are retrieved.
+    os.environ['DISPLAY'] = ''
+
+    # Create the temporary credentials file, and set the Chip to use it.
+    tmp_creds = '.test_remote_cfg'
+    with open(tmp_creds, 'w') as tmp_cred_file:
+        tmp_cred_file.write(json.dumps({'address': 'localhost', 'port': 8081}))
+    gcd_chip.set('option', 'remote', True)
+    gcd_chip.set('option', 'credentials', os.path.abspath(tmp_creds))
```

Comment:
Using a local test fixture might be helpful here to reduce some of this duplicated setup (and improve the clarity of the tests).

I think this can wait for a future PR next time you happen to touch this file, though.

Reply:
That's a good idea - and it's probably better to do it now. I think we should try to be more proactive about test hygiene with the higher volume of changes that we're dealing with.